lib/rules/valid-v-if-template-root.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * @author Perry Song
+ * @copyright 2023 Perry Song. All rights reserved.
+ * See LICENSE file in root directory for full license.
+ */
+'use strict'
+
+const utils = require('../utils')
+
+/**
+ * Get the number of root element directive
+ * @param {VNode[]} rootElements The start tag node to check.
+ * @param {string} directiveName The directive name to check.
+ */
+function getDirectiveLength(rootElements, directiveName) {
+  if (!directiveName) return 0
+  return rootElements.filter(
+    (element) =>
+      element.type === 'VElement' && utils.hasDirective(element, directiveName)
+  ).length
+}
+
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'enforce valid `v-if` directives on root element',
+      categories: ['vue3-essential', 'essential'],
```

Comment:
This would be a breaking change. Also, I don't think that everyone will like to forbid `v-if` on the root tag.